### PR TITLE
feat: Adding transport rtt to UI stats

### PR DESCRIPTION
**Pull request by SamuelBellomo**
### Description (*)
It was commented that RTT seemed high. Adding UTP side RTT to trigger conversation and show the difference.

### Issue Number(s) (*)
[MTT-2778](https://jira.unity3d.com/browse/MTT-2778)

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs [hunk 1]
```diff
@@ -25,21 +54,17 @@ public class NetworkStats : NetworkBehaviour
...
         // of attributes floating around.
 
-        public float LastRTT { get; private set; }
+        MovingWindowAverage BossRoomRTT = new MovingWindowAverage();
+        MovingWindowAverage UTP_RTT = new MovingWindowAverage();
 
-        [SerializeField]
-        [Tooltip("The interval to send ping RPCs to calculate the RTT. The bigger the number, the less reactive the stat will be to RTT changes")]
-        float m_PingIntervalSeconds = 0.1f;
+        const float m_PingIntervalSeconds = 0.1f;
```
Comment (LPLafontaineB) [inline Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs:60]:
```suggestion
        const float k_PingIntervalSeconds = 0.1f;
```

File: Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs [hunk 2]
```diff
@@ -16,6 +16,35 @@ namespace Unity.Multiplayer.Samples.BossRoom
     [RequireComponent(typeof(NetworkObject))]
     public class NetworkStats : NetworkBehaviour
     {
+        private class MovingWindowAverage
+        {
+            public float LastRTT { get; private set; }
+            Queue<float> m_MovingWindow = new Queue<float>();
+            const int k_MaxWindowSizeSeconds = 3; // it should take x seconds for the value to react to change
+            float m_MaxWindowSize => k_MaxWindowSizeSeconds / m_PingIntervalSeconds;
```
Comment (LPLafontaineB) [inline Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs:24]: Should this be a const instead?

**Review by LPLafontaineB — COMMENTED** [review]
nitpicks aside, lgtm

File: Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs (hunk 1 above)
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs]: done

File: Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs (hunk 2 above)
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs]: done

File: Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs [hunk 3]
```diff
@@ -16,6 +16,35 @@ namespace Unity.Multiplayer.Samples.BossRoom
     [RequireComponent(typeof(NetworkObject))]
     public class NetworkStats : NetworkBehaviour
     {
+        private class MovingWindowAverage
```
Comment (LukeStampfli) [inline Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs:19]: For a value like rtt I'd suggest to use an exponential moving average as it's a better indication of the current rtt and fluctuates less. So something like:
```cs
    public struct FloatEmaCalculator
    {
        private readonly float m_Alpha;
        private float m_Average;

        public float Average => m_Average;

        public FloatEmaCalculator(int lookBack, float average)
        {
            m_Alpha = 2f / (lookBack + 1);
            m_Average = average;
        }

        public float NextValue(float value) => m_Average = (value - m_Average) * m_Alpha + m_Average;
    }
```

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs]: agreed, done

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**